test_conformance/vulkan/test_vulkan_platform_device_info.cpp

```diff
@@ -48,6 +50,41 @@ _info device_info_table[] = {
 #undef STRING
 };
 
+/**
+ * Helper to return a string containing platform information for the specified
+ * platform info parameter.
+ */
+std::string get_platform_info_string(cl_platform_id platform,
+                                     cl_platform_info param_name)
+{
+    size_t size = 0;
+    int err;
+
+    if ((err = clGetPlatformInfo(platform, param_name, 0, NULL, &size))
+            != CL_SUCCESS
+        || size == 0)
+    {
+        throw std::runtime_error("clGetPlatformInfo failed\n");
+    }
+
+    std::vector<char> info(size);
+
+    if ((err = clGetPlatformInfo(platform, param_name, size, info.data(), NULL))
+        != CL_SUCCESS)
+    {
+        throw std::runtime_error("clGetPlatformInfo failed\n");
+    }
+
+    /* The returned string does not include the null terminator. */
+    return std::string(info.data(), size - 1);
+}
+
+bool is_platform_extension_available(cl_platform_id platform, const char* extensionName)
```

Comment:
These new functions really ought to go into the harness so they can be reused by other tests.  I'm a little surprised we don't have them already...

Reply:
Sure.